ext/nnstreamer/tensor_decoder/tensordec-boundingbox.cc

```diff
@@ -0,0 +1,947 @@
+      } else {
+        label = g_strdup_printf ("%s", labeldata.labels[a->class_id]);
+      }
+
+      label_len = strlen (label);
+      /* x1 is the same: x1 = MAX (0, (width * a->x) / i_width); */
+      y1 = MAX (0, (y1 - 14));
+      pos1 = &frame[y1 * width + x1];
+      for (j = 0; j < label_len; j++) {
+        unsigned int char_index = label[j];
+        if ((x1 + 8) > (int) width)
+          break; /* Stop drawing if it may overfill */
+        pos2 = pos1;
+        for (y2 = 0; y2 < 13; y2++) {
+          /* 13 : character height */
+          for (x2 = 0; x2 < 8; x2++) {
+            /* 8: character width */
+            *(pos2 + x2) = singleLineSprite[char_index][y2][x2];
+          }
+          pos2 += width;
+        }
+        x1 += 9;
+        pos1 += 9; /* charater width + 1px */
+      }
+    }
+  }
+}
+
+/**
+ * @brief Log the given results
+ */
+void
+BoundingBox::logBoxes (GArray *results)
+{
+  guint i;
+
+  nns_logi ("Detect %u boxes in %u x %u input image", results->len,
+      bdata->getInputWidth (), bdata->getInputHeight ());
+  for (i = 0; i < results->len; i++) {
+    detectedObject *b = &g_array_index (results, detectedObject, i);
+    if (labeldata.total_labels > 0)
+      nns_logi ("[%s] x:%d y:%d w:%d h:%d prob:%.4f",
+          labeldata.labels[b->class_id], b->x, b->y, b->width, b->height, b->prob);
+    else
+      nns_logi ("x:%d y:%d w:%d h:%d prob:%.4f", b->x, b->y, b->width, b->height, b->prob);
+  }
+}
+
+/**
+ * @brief Check the label relevant properties are valid
+ */
+gboolean
+BoundingBox::checkLabelProps ()
+{
+  if ((!label_path) || (!labeldata.labels) || (labeldata.total_labels <= 0))
+    return FALSE;
+  return TRUE;
+}
+
+/**
+ * @brief Set mode of bounding box
+ */
+int
+BoundingBox::setBoxDecodingMode (const char *param)
+{
+  bounding_box_modes previous = mode;
+
+  if (NULL == param || *param == '\0') {
+    GST_ERROR ("Please set the valid mode at option1 to set box decoding mode");
+    return FALSE;
+  }
+
+  mode = static_cast<bounding_box_modes> (find_key_strv (bb_modes, param));
+
+  if (mode != previous && mode != BOUNDING_BOX_UNKNOWN) {
+    if (previous != BOUNDING_BOX_UNKNOWN) {
+      delete bdata;
+    }
+
+    switch (mode) {
+      case MOBILENET_SSD_BOUNDING_BOX:
+      case OLDNAME_MOBILENET_SSD_BOUNDING_BOX:
+        bdata = new MobilenetSSD ();
+        break;
+      case MOBILENET_SSD_PP_BOUNDING_BOX:
+      case OLDNAME_MOBILENET_SSD_PP_BOUNDING_BOX:
+        bdata = new MobilenetSSDPP ();
+        break;
+      case OV_PERSON_DETECTION_BOUNDING_BOX:
+      case OV_FACE_DETECTION_BOUNDING_BOX:
+        bdata = new OVDetection ();
+        break;
+      case YOLOV5_BOUNDING_BOX:
+        bdata = new YoloV5 ();
+        break;
+      case YOLOV8_BOUNDING_BOX:
+        bdata = new YoloV8 ();
+        break;
+      case MP_PALM_DETECTION_BOUNDING_BOX:
+        bdata = new MpPalmDetection ();
+        break;
+      default:
```

Comment:
This creates dependency from the base class to its derived classe.

You may let the derived classes register required data and function callbacks for this switch-case statements.

e.g.,
```
class derived_class : public boxproperty
{
  friend class registerer;
  class registerer {
    registerer() { /* register a function callback doing "return new derived_class();" */ }
  };

  static registerer register;
};
```
in the derived .cc file.

Reply:
I misunderstood the previous comment because it was an unfamiliar concept.
Now base class do not need to be modifed when new derived class added or changed.